wrapper/src/main/java/software/amazon/jdbc/util/CoreServicesContainer.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc.util;
+
+import software.amazon.jdbc.util.events.BatchingEventPublisher;
+import software.amazon.jdbc.util.events.EventPublisher;
+import software.amazon.jdbc.util.monitoring.CoreMonitorService;
+import software.amazon.jdbc.util.monitoring.CoreMonitorServiceImpl;
+import software.amazon.jdbc.util.storage.StorageService;
+import software.amazon.jdbc.util.storage.StorageServiceImpl;
+
+/**
+ * A singleton container object used to instantiate core services. This class should be used instead of directly
+ * instantiating core services so that only one instance of each service is instantiated.
+ */
+public class CoreServicesContainer {
+  private static final CoreServicesContainer INSTANCE = new CoreServicesContainer();
```

Comment:
The instance will initialize immediately when the class is loaded. Is it desirable behaviour? Technically it's a singleton but with no delayed/lazy initialization.